### ISSUE TITLE
Working locally, failing in the CI (Jenkins)

### ISSUE THREAD

**Issue by dcacenabes** [DataDog/dd-sdk-swift-testing#44]
### Fails in Jenkins

Thinks are working for me locally with version 0.9.4 (integrated using Cocoapods and configured via scheme at the moment), but in Jenkins tests are crashing 😕 
We have three test targets and got the following errors:

```
Testing failed:
	XXXXXTests:
		App (17710) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Test crashed with signal ill before starting test execution. If you believe this error represents a bug, please attach the result bundle at …/xx.xcresult))
	YYYYYTests:
		xctest (17753) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Test crashed with signal ill before starting test execution. If you believe this error represents a bug, please attach the result bundle at …/xx.xcresult))
	ZZZZZTests:
		xctest (17784) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Test crashed with signal ill before starting test execution. If you believe this error represents a bug, please attach the result bundle at …/xx.xcresult))

```

I know this is very little information. I downloaded the xcresult but it just shows the same information. Is there a way that we can enable more verbose logging from the library so that we can troubleshoot better?

**Comment by nachoBonafonte** [comment]
I have created a beta version that might solve the issue if it is crashing in the framework at loading, please give it a try and let me know if it solves the issue. Version is `0.9.5-beta.1`, and is also available in cocoapods.

> In the VM I booted manually, running Big Sur 11.3 and Xcode 12.5, I got the same error!
Were you able to debug the issue inside Xcode? Was it possible to recover any crash report from the issue, or in the Console app inside the VM machine?

If it is still failing with this version I would need to know the Jenkins and VM setup you are using to try to reproduce it on my side. 

Thanks a lot

**Comment by nachoBonafonte** [comment]
Just one last thing I just thought about, are you setting the Jenkins environment variables to the test target, as documented [here](https://docs.datadoghq.com/continuous_integration/setup_tests/swift/?tab=jenkins#ci-provider-environment-variables)?

Thanks

**Comment by dcacenabes** [comment]
Regarding the Jenkins env vars:
This should not be the problem, because with my latest experiments Jenkins error was out of the equation (I manually booted a VM in Macstadium, cloned the project and built it there, reproducing the crash).

Regarding the  0.9.5-beta.1:
Now I am getting a different errors. I have 2 tests that are crashing. I am going to disable the testing library integration via DD_TEST_RUNNER env var and do another run to check if the tests now pass.

**Comment by nachoBonafonte** [comment]
Hi, do you have any news about it?. Could you find the reason of your crashes?

**Comment by dcacenabes** [comment]
Hi @nachoBonafonte , sorry for the long delay, I have been OOO. I've had to put this integration on hold due to other priorities. Unfortunately I never got to the bottom of the crashes. The crashes stack traces seemed unrelated to the datadog testing library, but if I disabled the DD library I did not get any crash.
If I get to allocate time to work on this again, would it be useful for you a stack trace?

**Comment by nachoBonafonte** [comment]
Yes please, it would be extremely useful to have those stack traces.

**Comment by dcacenabes** [comment]
Hi Ignacio,

Sorry for ghosting you. We were working on the Datadog integration as a side project and I had to jump work into something else.

I just had the chance to attempt to integrate the library again, and now it is working like a charm, both locally and in the CI, no crashes at all.

**Comment by nachoBonafonte** [comment]
Hi David, 
That's really good news!!!  Please do not hesitate in opening new issues if you find some issues or have any question.
